chalk-solve/src/recursive/fulfill.rs

```diff
@@ -173,11 +174,17 @@ impl<'s, I: Interner, Solver: SolveDatabase<I>, Infer: RecursiveInferenceTable<I
         let ProgramClauseImplication {
             consequence,
             conditions,
+            constraints,
             priority: _,
         } = fulfill
             .infer
             .instantiate_binders_existentially(fulfill.solver.interner(), clause);
```

Comment:
Hmm I suspect the problem is that somehow this is not instantiating the `^1` in the program clause constraints...